### PR TITLE
Remove `GetDomain` functions on assorted method/type/assembly classes which are always in the single app domain

### PR DESCRIPTION
**Pull request by elinor-fung**
- Remove `GetDomain` functions on `MethodDesc`, `MethodTable`, `TypeDesc`, and `TypeHandle`. They always return the single current app domain.
- Remove `GetDomain` functions and variables on  `Assembly`, `DomainAssembly`, and `Module`. The actual domain they are constructed with is always the single current app domain.

### CODE REVIEW

**Review by AaronRobinsonMSFT — COMMENTED** [review]
There are several places where `BaseDomain::IsAppDomain()` is called. Ideally `BaseDomain` would be an abstract class so that is fine, but what about the `SystemDomain`. Did we confirm that `SystemDomain` is never cast to a `BaseDomain` and passed around?

File: src/coreclr/vm/jithelpers.cpp [hunk 1]
```diff
@@ -3354,20 +3354,9 @@ CORINFO_GENERIC_HANDLE JIT_GenericHandleWorker(MethodDesc * pMD, MethodTable * p
...
-        }
-        else
-        {
-            pDictDomain = pMD->GetDomain();
-        }
-
         // Add the normalized key (pDeclaringMT) here so that future lookups of any
         // inherited types are faster next time rather than just just for this specific pMT.
-        JitGenericHandleCacheKey key((CORINFO_CLASS_HANDLE)pDeclaringMT, (CORINFO_METHOD_HANDLE)pMD, signature, pDictDomain);
+        JitGenericHandleCacheKey key((CORINFO_CLASS_HANDLE)pDeclaringMT, (CORINFO_METHOD_HANDLE)pMD, signature, AppDomain::GetCurrentDomain());
```
Comment (jkotas) [inline src/coreclr/vm/jithelpers.cpp:3359]: The domain argument of `JitGenericHandleCacheKey` ctor can be deleted.

**Comment by elinor-fung** [comment]
> Did we confirm that SystemDomain is never cast to a BaseDomain and passed around?

I checked this for the things I touched.

**Review by jkotas — APPROVED** [review]
Thanks

**Review by AaronRobinsonMSFT — APPROVED** [review]

**Pull request merged**